### PR TITLE
AO3-4619 - Add Github documentation and templates

### PR DESCRIPTION
**Pull request by ariana-paris**
## Fixes

Fixes https://otwarchive.atlassian.net/browse/AO3-4619

## Purpose

Add templates to discourage issues and standardise pull requests, and CONTRIBUTING document so our Github repo looks like other Open Source repos.

## Testing

See notes on [AO3-4619](https://otwarchive.atlassian.net/browse/AO3-4619)

## References

Discussed briefly in chat.

## Credit

What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](http://archiveofourown.org/admin_posts?tag=1)?

Ariana

What pronouns do you prefer (she/her, he/him, zie/hir etc)?

she/her

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Please do not create issues here.
+
+## To get help
+
+If you need help using [Archive of Our Own](http://archiveofourown.org), please [contact its Support team](http://archiveofourown.org/support)
+
+## To suggest a new feature
+
+[AO3 Support](http://archiveofourown.org/support) also track feature requests.
```
Comment (cjrecordvt) [inline .github/ISSUE_TEMPLATE.md]: "tracks"

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# We do not use Github for issues!
+
+Please do not create issues here.
```
Comment (cjrecordvt) [inline .github/ISSUE_TEMPLATE.md]: Since we list it below, do we want to link to JIRA, here?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+### Resolve the issue
+
+Consult our [Developer Documentation](https://github.com/otwcode/otwarchive/wiki) on how to set up a local development 
+environment so you can work on the Archive code.
+
+### Submit your pull request
+
+Please read the checklist on [our pull request template](https://github.com/otwcode/otwarchive/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
+on the steps you need to perform to make we can accept your pull request.
```
Comment (sarken) [inline CONTRIBUTING.md]: Typo -- should be "make sure"

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# We do not use Github for issues!
```
Comment (sarken) [inline .github/ISSUE_TEMPLATE.md]: It should be "GitHub" here and throughout

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (sarken) [inline .github/ISSUE_TEMPLATE.md]: I think that would make sense, and maybe mention that only OTW volunteers should do it, and everyone else should report bugs to Support?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+* [ ] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
+* [ ] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
+* [ ] Have you added tests for any changed functionality?
+* [ ] Have you added the JIRA issue number as the *first* thing in your pull request title (eg: `AO3-1234 Fix thing`)
+* [ ] Have you updated the JIRA issue with the information below?
+
+# Helpful things
+
+## Fixes
```
Comment (sarken) [inline .github/PULL_REQUEST_TEMPLATE.md]: Could we get rid of the word and heading fixes and keep it so the URL is the very first thing in the PR message?

File: CONTRIBUTING.md (hunk 3 above)
Comment (sarken) [inline CONTRIBUTING.md]: Aslo, should that be "for the steps you need to perform?"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Fixes https://otwarchive.atlassian.net/browse/AO3-**XXXX** (please fill in issue number and remove this comment)
+
+## Purpose
+
+What does this PR do?
+
+## Testing
+
+How can the Archive's QA team verify that this is work as you intended? (If you have access, please copy this into the JIRA ticket for them!)
```
Comment (sarken) [inline .github/PULL_REQUEST_TEMPLATE.md]: "this is working" or "this works"

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+### Testing and fixing bugs
+
+Once your pull request is merged, it will be deployed to our internal Staging server and our manual QA team will check it.
+We may ask you to make further changes and submit a new pull request if we find any problems that will affect our users'
+ability to benefit from your changes. 
+
+### Release
+
+Once we have checked that all is well, your contribution will be deployed to [Archive of Our Own](http://archiveofourown.org) and you will be credited
+in the [AO3 release notes](http://archiveofourown.org/admin_posts?tag=1)! Make sure you tell us your preferred name and pronoun
```
Comment (sarken) [inline CONTRIBUTING.md]: We capitalize Release Notes in the pull request template above (`What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](http://archiveofourown.org/admin_posts?tag=1)?`) -- should we do the same here?

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (ariana-paris) [inline .github/ISSUE_TEMPLATE.md]: I'll change that to say "The AO3 Support team" as in my head, "AO3 Support" is potentially a collective noun, and "team" is unambiguous.

Comment (sarken) [inline .github/ISSUE_TEMPLATE.md]: I think this is an American/British English thing -- American English would use the singular verb "tracks" with a collective noun like "team." Wikipedia seems to confirm my suspicions https://en.wikipedia.org/wiki/Collective_noun

**Comment by sarken** [comment]
When you add the readme changes mumble proposed, please tweak this part: `You can find more information about the history and future of the AO3 project [on the OTW website](http://transformativeworks.org/projects/archive).`

The part that's link-ified should probably be "information about the history and future of the AO3" or even just "the history and future of the AO3," since "on the OTW website" isn't very descriptive. If that's a little awkward, then maybe reword the whole thing to: `"The OTW website has [more information about the history and future of the AO3 project](http://transformativeworks.org/projects/archive)."`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# Contributing to Archive of Our Own
```
Comment (sarken) [inline CONTRIBUTING.md]: Should this be OTW-Archive?

Comment (ariana-paris) [inline CONTRIBUTING.md]: Since we're not actually producing a piece of software called "OTW-Archive" but rather a website called "Archive of Our Own", imo, the answer is no. People who want to contribute will be contributing to the website.
Obviously, I'll change it if anyone feels strongly that it should be "OTW-Archive" though!

Comment (sarken) [inline CONTRIBUTING.md]: Sounds good to me! I just wanted to check given we're still talking about OTW-Archive software and the Archive separately in the readme.
